Fix date rendering

rpeterman-gp opened the pull request:
- add DateTime component to format date using toLocaleDateString
- update components to use DateTime component
- update conversions of broadcastDate to Date to append midnight time to date string
- remove moment and moment-timezone

## To Review

- [x] Checkout Branch.
- [x] Update .env to use live API: `API_URL_BASE=https://api.theworld.org`
- [x] Run `yarn`.
- [x] Run `yarn dev`.
- [x] Go to [localhost:3000](https://localhost:3000).

> ...then...

- [x] Ensure dates in all cards are correct.
- [x] Refresh page and ensure wrong date is not flashed momentarily
- [x] View a story and ensure dates are correct
- [x] View a episode and ensure dates are correct
- [x] View a segment and ensure dates are correct
- [x] Play some audio and ensure correct date is shown in player
- [x] Search for `olympics` and ensure dates on cards for stories, episodes, and segments are correct

rpeterman-gp committed:
fix: date rendering

- add DateTime component to format date using toLocaleDateString
- update components to use DateTime component
- update conversions of broadcastDate to Date to append midnight time to date string
- remove moment and moment-timezone

kookster commented on components/DateTime/DateTime.tsx:
```diff
@@ -0,0 +1,39 @@
+      : date;
+  const renderDate =
+    usedDateValue instanceof Date ? usedDateValue : new Date(usedDateValue);
+  const formattedDate = renderDate.toLocaleDateString(
```
so, this will render for the local time zone? Will this change based on the client/browser timezone, or always the same based on SSR?

brandonhundt reviewed (approved):
:shipit: Thanks @rpeterman-gp I see you swapped out the date component. LGTM

rpeterman-gp committed:
feat(app): add settings to app data

- update DateTime to use app timezone setting
- allow dates in audio data info array
- use DateTime to render audio info dates

rpeterman-gp committed:
feat(datetime): ensure iso 8601 date string have time and offset

kookster commented on components/DateTime/DateTime.tsx:
```diff
@@ -0,0 +1,68 @@
+          ...(timeZone && { timeZone }),
+          timeZoneName: 'longOffset'
+        })
+        .match(/GMT(.+)/)?.[1];
```
this works, and the tz hour is formatted for what you need
I wondered about [getTimezoneOffset()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset) ?
```
const tz = `${-(new Date().getTimezoneOffset() / 60)}:00`
```
but then you'd need to 0 pad the hour, and format it - not sure it's worth it.

kookster commented on lib/fetch/app/fetchGqlApp.ts:
```diff
@@ -56,6 +60,7 @@ const GET_APP = gql`
 
 export const fetchGqlApp = async () => {
   const response = await gqlClient.query<{
+    allSettings: Settings;
```
I see, this is the method where you get the global settings you were telling me about.

kookster reviewed (approved):
lgtm!
